src/WebJobs.Script/Workers/Profiles/WorkerProfileManager.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Script.Workers.Profiles;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers
+{
+    internal class WorkerProfileManager : IWorkerProfileManager
+    {
+        private readonly ILogger _logger;
+        private readonly IEnumerable<IWorkerProfileConditionProvider> _conditionProviders;
+        private Dictionary<string, List<WorkerDescriptionProfile>> _profiles = new Dictionary<string, List<WorkerDescriptionProfile>>();
+        private string _activeProfile;
+
+        public WorkerProfileManager(ILogger logger,
+                                             IEnumerable<IWorkerProfileConditionProvider> conditionProviders)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _conditionProviders = conditionProviders ?? throw new ArgumentNullException(nameof(conditionProviders));
+            _activeProfile = string.Empty;
+        }
+
+        public void SaveWorkerDescriptionProfiles(List<WorkerDescriptionProfile> workerDescriptionProfiles, string language)
+        {
+            _profiles.Add(language, workerDescriptionProfiles);
+        }
+
+        private bool GetEvaluatedProfile(string language, out WorkerDescriptionProfile evaluatedProfile)
+        {
+            if (_profiles.TryGetValue(language, out List<WorkerDescriptionProfile> profiles))
+            {
+                foreach (var profile in profiles)
+                {
+                    if (profile.EvaluateConditions())
+                    {
+                        evaluatedProfile = profile;
+                        return true;
+                    }
+                }
+            }
+            evaluatedProfile = null;
+            return false;
+        }
+
+        public void LoadWorkerDescriptionFromProfiles(RpcWorkerDescription defaultWorkerDescription, out RpcWorkerDescription workerDescription)
+        {
+            if (GetEvaluatedProfile(defaultWorkerDescription.Language, out WorkerDescriptionProfile profile))
+            {
+                _logger?.LogInformation($"Worker initialized with profile - {profile.Name}, Profile ID {profile.ProfileId} from worker config.");
+                _activeProfile = profile.ProfileId;
+                workerDescription = profile.ApplyProfile(defaultWorkerDescription);
+            }
+            workerDescription = defaultWorkerDescription;
```

Comment:
it looks like we're still setting the default profile here?

`workerDescription = profile.ApplyProfile(defaultWorkerDescription);`

- Here we take the `defaultWorkerDescription` variable and override it with properties from the profile 

`workerDescription = defaultWorkerDescription;`

- Then when we are done with the if condition, we override `workerDescription` with the `defaultWorkerDescription` from the method parameters.

So it seems like we never actually use the overridden description with profile properties applied?

It probably makes sense to exit the method after line 55 here